source/tutorials/broadcom.rst

```diff
@@ -0,0 +1,203 @@
+       MAKE="make KBASE=/lib/modules/${kernelver}"
+       CLEAN="make KBASE=/lib/modules/${kernelver} clean"
+       BUILT_MODULE_NAME=wl
+       DEST_MODULE_LOCATION=/kernel/drivers/net/wireless
+       AUTOINSTALL=yes
+       EOF
+
+#. Run the :command:`dkms` commands to add the broadcom-wl module to the dkms
+   tree, build it, and install it. This example uses version *6.30.223.271*.
+
+   .. code-block:: bash
+
+      dkms add -m broadcom-wl -v 6.30.223.271
+      dkms build -m broadcom-wl -v 6.30.223.271
+      dkms install -m broadcom-wl -v 6.30.223.271
+
+#. Blacklist all other variations of Broadcom drivers from loading to prevent
+   conflicts and problems.
+
+   .. code-block:: bash
+
+      sudo mkdir -p /etc/modprobe.d/
+
+      sudo tee /etc/modprobe.d/broadcom.conf > /dev/null <<'EOF'
+      blacklist b43
+      blacklist b43legacy
+      blacklist ssb
+      blacklist bcm43xx
+      blacklist brcm80211
+      blacklist brcmfmac
+      blacklist brcmsmac
+      blacklist bcma
+      EOF
+
+#. Reboot the system and check that the module is loaded and working. If not,
+   try manually updating dependencies and loading the module.
+
+   .. code-block:: bash
+
+      depmod -a
+      modprobe wl
+
+
+Firmware
+********
+
+In addition to device drivers, devices require firmware that gets loaded onto
+the device directly.
+
+Firmware for Broadcom devices are not fully open-source and not always
+licensed for redistribution. |CL| kernel bundles :ref:`include the
+linux-firmware bundle <firmware-included-begin>` which contains the firmware
+binaries that are able to be redistributed. If your device's firmware is part
+of the linux-firmware repository, nothing else is needed. This is usually
+the case for devices supported by the *brcmfmac* and *brcmsmac* drivers.
+
+In other cases, firmware may need to be obtained or extracted manually from a
+trusted source because it is not licensed for distribution. This is usually
+the case for devices supported by the *b43* and *b43legacy* drivers. Obtaining
+these firmware is out of scope for this document, however there is
+information about solutions to this problem is available on the `Linux
+wireless wiki
+<http://linuxwireless.sipsolutions.net/en/users/Drivers/b43/#firmware>`_.
+
+On |CL| systems, firmware should be placed in :file:`/etc/firmware`. See the
+:ref:`firmware` documentation for more information on loading custom
+firmware.
+
+
+Troubleshooting
+***************
+
+- See which drivers are currently loaded with the :command:`lsmod` and
+  :command:`modinfo` commands.
+
+- If you're device is not showing up or having intermittent issues, ensure the
+  card is not blocked by the kernel with the :command:`rfkill` command.
+
+- Try blacklisting all the other variations of drivers not intended to be
+  used. In some cases, the wrong device driver will be loaded causing
+  problems. 
+
+- If an external firmware is required, it may be trying to load from a
```

Comment:
```suggestion
- If external firmware is required, it may be trying to load from a
```
or
```suggestion
- If an external firmware image is required, it may be trying to load from a
```

Reply:
@bwarden.  Thanks a lot for your review. And how!